Remove fallback loading of community auth manager

henriknyman opened the pull request:
This fixes a bug where the auth manager service loading would fall back on community auth manager if an exception was thrown during instantiation.
- Moves all security setup into a `SecurityModule` that manages `AuthManager`, `SecurityLog` and auth procedures.
- Removes configurable auth manager loading from community edition
- Replaces it with configurable security module loading in enterprise edition
- Fails hard if the configured security module could not be instantiated

OliviaYtterbrink committed:
Create all security in security module

henriknyman committed:
Make security module configurable in enterprise

Removed `unsupported.dbms.security.auth_manager` from community
and replaced it with `unsupported.dbms.security.module` in enterprise.
This allows third parties to deploy their own security module through service
loading in enterprise edition.

henriknyman committed:
Move creation of auth disabled auth manager out of Security module

- Fixed EditionModuleTest

henriknyman committed:
Move log and settings into security module

Move the `SecurityLog` and `SecuritySettings` into separate sub-packages
of the enterprise security module.

Mats-SX commented on enterprise/kernel/src/main/java/org/neo4j/kernel/impl/enterprise/EnterpriseEditionModule.java:
```diff
@@ -85,7 +87,20 @@ protected StatementLocksFactory createStatementLocksFactory( Locks locks, Config
+        }
+        else
+        {
+            platformModule.life.add( platformModule.dependencies.satisfyDependency( EnterpriseAuthManager.NO_AUTH ) );
```
Nice! This way we do no service loading at all if auth is disabled, right?

henriknyman commented on community/security/src/main/java/org/neo4j/server/security/auth/AuthProcedures.java:
```diff
@@ -48,7 +51,8 @@ public void createUser(
     {
         BasicAuthSubject subject = BasicAuthSubject.castOrFail( authSubject );
-        subject.getAuthManager().newUser( username, password, requirePasswordChange );
+//        subject.getAuthManager().newUser( username, password, requirePasswordChange );
```
This will be removed by 46a3d6b9c7632293b128d271c3475ebae878dec3

henriknyman committed:
Register user management procedures conditionally

Only register user management procedures if native realm is enabled.

- Divide enterprise security procedures into always available `SecurityProcedures`,
and `UserManagementProcedures` that requires a `UserManager`.

henriknyman committed:
changeUserPassword procedure should not kick out user

The `changeUserPassword` now does not terminate active transactions and
connections for the user.

craigtaverner commented:
This PR is incorporated into https://github.com/neo4j/neo4j/pull/8083

henriknyman commented:
This was merged together with https://github.com/neo4j/neo4j/pull/8083
